### PR TITLE
EN-10372: general refactoring

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
Removed unused components + fixes

### CODE REVIEW

File: core/accumulator/timeAccumulator_test.go [hunk 1]
```diff
@@ -5,9 +5,11 @@ import (
 	"testing"
 	"time"
 
-	"github.com/ElrondNetwork/elrond-go/core"
-	"github.com/ElrondNetwork/elrond-go/core/accumulator"
-	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	core2 "github.com/ElrondNetwork/elrond-go-core/core"
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply core/accumulator/timeAccumulator_test.go:9]: fixed

File: core/accumulator/timeAccumulator_test.go [hunk 2]
```diff
@@ -17,7 +19,7 @@ var timeout = time.Second * 2
 func TestNewTimeAccumulator_InvalidMaxWaitTimeShouldErr(t *testing.T) {
 	t.Parallel()
 
-	ta, err := accumulator.NewTimeAccumulator(accumulator.MinimumAllowedTime-1, 0)
+	ta, err := accumulator.NewTimeAccumulator(accumulator.MinimumAllowedTime-1, 0, &mock.LoggerFake{})
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply core/accumulator/timeAccumulator_test.go:22]: renamed

File: core/appStatusPolling/appStatusPolling.go [hunk 3]
```diff
@@ -5,34 +5,37 @@ import (
 	"sync"
 	"time"
 
-	logger "github.com/ElrondNetwork/elrond-go-logger"
-	"github.com/ElrondNetwork/elrond-go/core"
-	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	core2 "github.com/ElrondNetwork/elrond-go-core/core"
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply core/appStatusPolling/appStatusPolling.go:9]: fixed

File: core/appStatusPolling/appStatusPolling_test.go [hunk 4]
```diff
@@ -5,45 +5,52 @@ import (
 	"testing"
 	"time"
 
-	"github.com/ElrondNetwork/elrond-go/core"
-	"github.com/ElrondNetwork/elrond-go/core/appStatusPolling"
-	"github.com/ElrondNetwork/elrond-go/core/mock"
-	"github.com/ElrondNetwork/elrond-go/statusHandler"
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	core2 "github.com/ElrondNetwork/elrond-go-core/core"
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply core/appStatusPolling/appStatusPolling_test.go:9]: fixed

File: core/appStatusPolling/appStatusPolling_test.go [hunk 5]
```diff
@@ -5,45 +5,52 @@ import (
...
-	_, err := appStatusPolling.NewAppStatusPolling(&statusHandler.NilStatusHandler{}, time.Duration(-1))
+	_, err := appStatusPolling.NewAppStatusPolling(&mock.StatusHandlerFake{}, time.Duration(-1), core2.NewConsoleLogger())
 	assert.Equal(t, err, appStatusPolling.ErrPollingDurationToSmall)
 }
 
 func TestNewAppStatusPooling_ZeroPollingDurationShouldErr(t *testing.T) {
 	t.Parallel()
 
-	_, err := appStatusPolling.NewAppStatusPolling(&statusHandler.NilStatusHandler{}, 0)
+	_, err := appStatusPolling.NewAppStatusPolling(&mock.StatusHandlerFake{}, 0, core2.NewConsoleLogger())
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply core/appStatusPolling/appStatusPolling_test.go:39]: renamed

File: core/consoleLogger.go [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+// NewConsoleLogger returns a new instance of consoleLogger
+func NewConsoleLogger() *consoleLogger {
+	return &consoleLogger{}
+}
+
+// Trace will print a trace log
+func (c consoleLogger) Trace(message string, args ...interface{}) {
+	fmt.Printf("[TRACE] %s %v", message, getPrintableArgs(args))
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply core/consoleLogger.go:15]: done. :medal_sports:

File: core/file_test.go [hunk 7]
```diff
@@ -18,7 +17,7 @@ type TestStruct struct {
 }
 
 func TestOpenFile_NoExistingFileShouldErr(t *testing.T) {
-	t.Parallel()
+	// t.Parallel()
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply core/file_test.go:20]: fixed

File: core/loggingFunctions_test.go [hunk 8]
```diff
@@ -17,5 +17,5 @@ func TestDumpGoRoutinesToLogShouldNotPanic(t *testing.T) {
 		}
 	}()
 
-	DumpGoRoutinesToLog(0)
+	DumpGoRoutinesToLog(0, nil)
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply core/loggingFunctions_test.go:22]: ook..

File: core/mock/loggerFake.go [hunk 9]
```diff
@@ -0,0 +1,34 @@
+package mock
+
+// LoggerFake -
+type LoggerFake struct {
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply core/mock/loggerFake.go:4]: renamed

File: core/mock/marshalizerStub.go [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+
+// MarshalizerStub -
+type MarshalizerStub struct {
+	MarshalCalled   func(obj interface{}) ([]byte, error)
+	UnmarshalCalled func(obj interface{}, buff []byte) error
+}
+
+// Marshal -
+func (ms *MarshalizerStub) Marshal(obj interface{}) ([]byte, error) {
+	return ms.MarshalCalled(obj)
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply core/mock/marshalizerStub.go:11]: added

File: core/mock/statusHandlerFake.go [hunk 11]
```diff
@@ -0,0 +1,38 @@
+package mock
+
+// StatusHandlerFake -
+type StatusHandlerFake struct {
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply core/mock/statusHandlerFake.go:4]: renamed

File: core/consoleLogger.go [hunk 12]
```diff
@@ -0,0 +1,66 @@
+package core
+
+import "fmt"
+
+type consoleLogger struct {
+}
+
+// NewConsoleLogger returns a new instance of consoleLogger
+func NewConsoleLogger() *consoleLogger {
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply core/consoleLogger.go:9]: done

File: core/watchdog/watchdog.go [hunk 13]
```diff
@@ -5,34 +5,38 @@ import (
 	"runtime/pprof"
 	"time"
 
-	logger "github.com/ElrondNetwork/elrond-go-logger"
-	"github.com/ElrondNetwork/elrond-go/core"
-	"github.com/ElrondNetwork/elrond-go/core/check"
-	"github.com/ElrondNetwork/elrond-go/data/endProcess"
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	core2 "github.com/ElrondNetwork/elrond-go-core/core"
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply core/watchdog/watchdog.go:9]: fixed

File: core/loggingFunctions.go [hunk 14]
```diff
@@ -7,15 +7,28 @@ import (
...
 
 // DumpGoRoutinesToLog will print the currently running go routines in the log
-func DumpGoRoutinesToLog(goRoutinesNumberStart int) {
-	buffer := new(bytes.Buffer)
-	err := pprof.Lookup("goroutine").WriteTo(buffer, 2)
+func DumpGoRoutinesToLog(goRoutinesNumberStart int, log Logger) {
+	if log == nil {
+		return
+	}
+	buf := new(bytes.Buffer)
```
Comment (miiu96) [inline core/loggingFunctions.go:14]: add empty line

File: core/watchdog/watchdog.go [hunk 15]
```diff
@@ -51,11 +54,11 @@ func (w *watchdog) defaultWatchdogExpiry(watchdogID string) {
 	buffer := new(bytes.Buffer)
 	err := pprof.Lookup("goroutine").WriteTo(buffer, 1)
 	if err != nil {
-		log.Error("could not dump goroutines")
+		w.log.Error("could not dump goroutines", "err", err)
```
Comment (miiu96) [inline core/watchdog/watchdog.go:57]: "error" instead of "err"

File: core/loggingFunctions.go (hunk 14 above)
Comment (bogdan-rosianu) [reply core/loggingFunctions.go:14]: done

File: core/loggingFunctions_test.go [hunk 16]
```diff
@@ -17,5 +19,12 @@ func TestDumpGoRoutinesToLogShouldNotPanic(t *testing.T) {
...
 
-	DumpGoRoutinesToLog(0)
+	DumpGoRoutinesToLog(0, &mock.LoggerMock{})
+}
+
+func TestGetRunningGoRoutines(t *testing.T) {
+	t.Parallel()
+
+	res := GetRunningGoRoutines(&mock.LoggerMock{})
+	require.NotNil(t, res)
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply core/loggingFunctions_test.go:29]: done

File: core/watchdog/watchdog.go (hunk 15 above)
Comment (bogdan-rosianu) [reply core/watchdog/watchdog.go:57]: done

**Review by miiu96 — DISMISSED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by miiu96 — APPROVED** [review]

**Pull request merged**